### PR TITLE
Pass host annotations when creating from factory

### PR DESCRIPTION
**Pull request by egvili**
### Desired Outcome

Add ability to pass annotations when creating host from  hostfactory

### Implemented Changes

Extend createHostData with one annotation field, and use that field when running create host command

### Connected Issue/Story

Resolves #[relevant GitHub issue(s), e.g. 76]

CyberArk internal issue link: [insert issue ID]()

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [ ] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [ ] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [ ] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [ ] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [ ] There are no security aspects to these changes

### CODE REVIEW

File: conjur_api/models/hostfactory/create_host_data.py [hunk 1]
```diff
@@ -19,17 +19,19 @@ class CreateHostData:
     def __init__(self,
                  # using id shadows the internal id
                  host_id: str = "",
-                 token: str = ""):
+                 token: str = "",
+                 annotations: dict = None):
```
Comment (rafis3) [inline conjur_api/models/hostfactory/create_host_data.py:23]:
```suggestion
                 annotations: dict = {}):
```
Would this work? And then you won't need the line above will be:
```python
self.annotations = annotations
```

File: tests/models/test_unit_hostfactory.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import unittest
+
+from conjur_api.models import CreateHostData
+
+
+class HostFactoryTest(unittest.TestCase):
+
+    def test_create_host_data_fields(self):
```
Comment (rafis3) [inline tests/models/test_unit_hostfactory.py:8]: Can we please add more tests? I don't think this single test covers the functionality that you added like:

- Creating a host and seeing that the host was created successfully with annotations
- Failing to create a host
- Adding multiple annotations to ensure that the dictionary translation works with multiple keys and values

Also can you please also add these same tests as integration tests as well?

**Review by rafis3 — COMMENTED** [review]
@egvili added some comments and requests. Please take a look

File: conjur_api/models/hostfactory/create_host_data.py (hunk 1 above)
Comment (egvili) [reply conjur_api/models/hostfactory/create_host_data.py:23]: That's what I did at first but it fails the lint. Apparently, if the default parameter changes in the function then the change will apply for the next time someone calls this function

File: tests/models/test_unit_hostfactory.py (hunk 2 above)
Comment (egvili) [reply tests/models/test_unit_hostfactory.py:8]: I will test the scenarios you described in unit and integration tests

File: tests/integration/test_integration_hostfactory.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        token: json = await c.create_token(CreateTokenData(host_factory="factory", hours=1, cidr="0.0.0.0/32"))
+        with self.assertRaises(HttpStatusError) as ex:
+            await c.create_host(CreateHostData("new-host", token[0]['token'], {"creator": "me", "date": "today"}))
+        self.assertEqual(401, ex.exception.status)
+
+
+    @classmethod
+    async def _add_test_data(cls):
+        c = await create_client("admin", os.environ['CONJUR_AUTHN_API_KEY'])
+        await c.load_policy_file("root", "tests/integration/policies/hostfactory.yml")
```
Comment (rafis3) [inline tests/integration/test_integration_hostfactory.py:41]: I think you are missing here an extra line. The lint might catch that

File: tests/integration/integ_utils.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    account = "dev"
+    conjur_data = ConjurConnectionInfo(
+        conjur_url=conjur_url,
+        account=account
+    )
+    credentials_provider = SimpleCredentialsProvider()
+    credentials = CredentialsData(username=username, password=password, machine=conjur_url)
+    credentials_provider.save(credentials)
+    return Client(conjur_data, credentials_provider=credentials_provider,
+                  ssl_verification_mode=SslVerificationMode.INSECURE)
```
[the remark this answers is not in the document]
Comment (egvili) [reply tests/integration/integ_utils.py:19]: The linter missed that, but I will add a newline anyway here and in other places

**Review by rafis3 — APPROVED** [review]
@egvili looks good :)
Notice just that there are two places that are missing an empty line in the end of the file. I think the Python lint is not going to like that

**Pull request merged**